### DISCUSSION TITLE
Ability to Specify CMS Logo in config

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9391] [Ideas]
> Created by: Nathaniel Hammond (nathanielhammond@gmail.com) on 2015/03/23 15:31:39 +0000
> Votes at time of UserVoice import: 4

-----------------------------------

A config variable to be able to define the logo in General Settings.

**Comment by mattandrews** [comment]
I could do with this. We deploy Craft via a blue/green strategy (eg. spin up new instances from source and then swap over to them), meaning any logo I upload via the control panel gets wiped out between deploys.

**Comment by brandonkelly** [comment]
Why not just commit the logo to Git and have it be part of the deploy?

**Comment by mattandrews** [comment]
Makes sense – is there a reference in the database somewhere to the filename Craft expects? I can see  a path like `/cpresources/4139eb4d/logo.svg?v=1548755111` being used, would it be safe to just hardcode a file at this path (or symlink one there) inside my repo?

**Comment by brandonkelly** [comment]
As long as there is a file in `storage/rebrand/logo/`, Craft will show it on the login page. No DB changes necessary.

**Comment by mattandrews** [comment]
Excellent! Thank you so much Brandon.

**Comment by FreekVR** [comment]
I still think it would be nice to be able to set this to a custom path -- or an environment variable. That way it's easy to have a different logo for local/test/staging/production for example as an indicator to users.

**Comment by uandco** [comment]
+1 for custom path.

Because the `rebrand` folder can only be in `storage` (or whatever the storage folder has been define to), it makes it very hard to deploy rebranding elements.

Our deployment process checks out files (including anything in "rebrand"), then mounts the persistent storage volume, essentially overwriting anything that sits within "storage".

Imho you should be able to set where the rebranding elements live, if you want them outside of the `storage` folder.

My current workaround is to symlink `storage/rebrand` but that's still not ideal as it needs to be done manually on the host after persistent storage is mounted, for any new environment I'm spawning.
